components/Roster/RosterItem/PendingRosterItem.vue

```diff
@@ -29,11 +35,16 @@
       },
     },
     methods: {
-      approve() {
-        this.$emit('approve', this.item);
+      async approve() {
+        this.menu = true;
+        if(await this.$refs.dialog.getResponse()) {
```

Comment:
Waits for user clicking on `I understand`, or not at all (if the user has already accepted) to accept the request.